pages/platform/kubernetes-k8s/cluster-autoscaler-example/guide.en-gb.md

```diff
@@ -0,0 +1,394 @@
+  conditions:
+    [...]
+  currentNodes: 3
+  observedGeneration: 2
+  upToDateNodes: 00            3        
+</code></pre>
+
+And in the `spec` section you can see that the `autoscale` parameter is set to `false`. In order to enable the autoscaler, you need to patch the node pool to set this field to `true`. Set also `desiredNodes` to 1, `minNodes` to 1 and `maxNodes` to 3, to be sure that we begin only with one active node and that we can grow the node pool up to 3 nodes.
+
+```bash
+kubectl patch nodepool &lt;your_nodepool_name> --type="merge" --patch='{"spec": {"autoscale": true, "desiredNodes": 1, "minNodes": 0, "maxNodes": 3}}'
+```
+
+As you can see in my example, patching the node pool definition enables the autoscaler:
+
+<pre class="console"><code>$ kubectl patch nodepool nodepool-b2-7 --type="merge" --patch='{"spec": {"autoscale": true}}'
+nodepool.kube.cloud.ovh.com/nodepool-b2-7 patched
+
+$ kubectl get nodepools
+NAME            FLAVOR   AUTO SCALED   [...]   ANTI AFFINITY   DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   MIN   MAX
+nodepool-b2-7   b2-7     true          [...]   false           1         1         0            1           0     3
+</code></pre>
+
+Now we can begin to deploy a test workload on the cluster.
+
+
+
+## Deploying a test workload
+
+In order to test the autoscaler, we propose you to install a  *Prime Numbers* Deployment that deploys several  instance of *Prime Numbers* pods. The *Prime Numbers* pod simply calculates prime numbers in the most performance ineffective way: by dividing every positive integer number by all the lower positive integers. It's a CPU intensive operation, but it use a minimal amount of memory.  
+
+> [!primary]
+>
+> The *Prime Numbers* pod is a small [NodeJS](https://nodejs.org/) programs. The main script, on `index.js`, is a [Express server](https://expressjs.com/) that launches a [worker thread](https://nodejs.org/api/worker_threads.html) were the prime number calculation is done outside the event loop.
+
+Create a `prime-number.yaml` manifest for the *Prime Numbers* deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prime-numbers
+spec:
+  selector:
+    matchLabels:
+      run: prime-numbers
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        run: prime-numbers
+    spec:
+      containers:
+      - name: prime-numbers
+        image: lostinbrittany/ovhcloud-k8s-autoscaling-demo_prime-numbers:0.0.1
+        resources:
+          limits:
+            cpu: 300m
+            memory: 30Mi
+          requests:
+            cpu: 150m
+            memory: 15Mi
+        ports:
+        - containerPort: 8080        
+```
+
+As you can see, we will begin by deploying 3 replicas of the pod. Each replica consommes 150m CPU (0.150 CPUs), and we are using B2-7 instances, with 2000m CPU (2 CPU_). In the tutorial we will increase the number of replicas to 12 then to 24, to see how the autoscaler grows up the node pool to 2 then to 3 nodes.
```

Comment:
maybe we should add an hint somewhere for the compatibility with the horizontal pod autoscaler to not scale replicas by hand

Reply:
Hmmm, we can add it later, I don't feel inspired right now to find the right phrase...
Or if you have it, please tell me :)